src/validators/object.ts

```diff
@@ -54,6 +54,11 @@ function validateObject<
   const validators = this.__values as Record<string, SomeSchema<any>>;
 
   const valueKeys = Object.keys(value);
+  const unknownKeysCount = valueKeys.filter((k) => !(k in validators)).length;
```

Comment:
`k in validators` checks for inherited properties too. Let's add a test for that:

```ts
const triangle = {a: 1, b: 2, c: 3};
function ColoredTriangle() {
  this.color = 'red';
}
ColoredTriangle.prototype = triangle;

const obj = new ColoredTriangle();
```